fa25-team-b/scripts/popularity_graph_generation.py

```
@@ -0,0 +1,912 @@
+                                'bedrooms': [],
+                                'age_restricted': []
+                            }
+
+                        if pd.notna(row.get('Bedrooms')):
+                            property_data[key]['bedrooms'].append(int(row['Bedrooms']))
+
+                        if pd.notna(row.get('Age Restricted')):
+                            property_data[key]['age_restricted'].append(str(row['Age Restricted']))
+
+            logging.info(f"Loaded property data from {file_path}: {len(property_data)} unique properties")
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)